functional/FunctionalTest.java

```diff
@@ -2742,17 +2745,25 @@ public static void deleteBucketPolicy() throws Exception {
     }
   }
 
-  public static void testSetBucketLifeCycle(String bucketName) throws Exception {
-    String lifeCycle =
-        "<LifecycleConfiguration><Rule><ID>expire-bucket</ID><Prefix></Prefix>"
-            + "<Status>Enabled</Status><Expiration><Days>365</Days></Expiration>"
-            + "</Rule></LifecycleConfiguration>";
-    client.setBucketLifeCycle(
-        SetBucketLifeCycleArgs.builder().bucket(bucketName).config(lifeCycle).build());
+  public static void testSetBucketLifecycle(String bucketName) throws Exception {
+    List<LifecycleRule> rules = new LinkedList<>();
+    rules.add(
+        new LifecycleRule(
+            Status.ENABLED,
+            null,
+            new Expiration((ZonedDateTime) null, 365, null),
+            new RuleFilter("logs/"),
+            "rule2",
+            null,
+            null,
+            null));
+    LifecycleConfiguration config = new LifecycleConfiguration(rules);
+    client.setBucketLifecycle(
+        SetBucketLifecycleArgs.builder().bucket(bucketName).config(config).build());
```

Comment:
Maybe we should call `getBucketLifecycle` also and verify that what was set is same as config passed in `setBucketLifecycle`

Reply:
I added into `getBucketLifecycle()` test